src/functional/iterator/embedded.py

```diff
@@ -534,7 +558,11 @@ def _make_tuple(
     if isinstance(field_or_tuple, tuple):
         return tuple(_make_tuple(f, indices) for f in field_or_tuple)
     else:
-        return field_or_tuple[indices]
+        data = field_or_tuple[indices]
+        if isinstance(data, np.ndarray):
+            return Column(0, data)
```

Comment:
This part is not clear to me. Can you give an example when this happens? I think a comment would be beneficial.

Reply:
Thanks for pointing that out. The implementation is not very nice, I wanted to think of a cleaner solution.

This is the case where no scan is involved but the fencil is executed in columns. Here we are packing the column slice of an input field into a `Column`. It should not be hidden in the tuple creation. Additionally, honestly I am not 100% sure if the 0 is correct here. I'll have to add a test.

Reply:
see comment below